src/main/java/org/broadinstitute/hellbender/engine/dataflow/transforms/RemoveDuplicateReadVariantPairs.java

```diff
@@ -0,0 +1,152 @@
+package org.broadinstitute.hellbender.engine.dataflow.transforms;
+
+import com.google.api.client.util.Sets;
+import com.google.cloud.dataflow.sdk.transforms.*;
+import com.google.cloud.dataflow.sdk.transforms.join.CoGbkResult;
+import com.google.cloud.dataflow.sdk.transforms.join.CoGroupByKey;
+import com.google.cloud.dataflow.sdk.transforms.join.KeyedPCollectionTuple;
+import com.google.cloud.dataflow.sdk.values.*;
+import com.google.common.collect.Lists;
+import com.google.common.collect.Maps;
+import org.broadinstitute.hellbender.exceptions.GATKException;
+import org.broadinstitute.hellbender.utils.read.GATKRead;
+import org.broadinstitute.hellbender.utils.variant.Variant;
+
+import java.util.*;
+
+
+/**
+ * RemoveDuplicateReadVariantPairs uses the UUID surrogates for GATKRead and Variant to remove duplicate KV pair values.
```

Comment:
This could use some clarification. After reading this line I still have no idea what this class does, except that its output is a subset of its input (and then reading the code, I see that I'm mistaken even in that).